extendr-api/src/robj/try_from_robj.rs

```diff
@@ -34,11 +34,7 @@ macro_rules! impl_try_from_scalar_integer {
                 // is problematic when converting a negative value to unsigned
                 // integer types (e.g. `-1i32 as u8` becomes 255).
                 if let Some(v) = robj.as_integer() {
-                    if let Ok(v) = Self::try_from(v) {
-                        return Ok(v);
-                    } else {
-                        return Err(Error::OutOfLimits(robj.clone()));
-                    }
+                    return Self::try_from(v).map_err(|_| Error::OutOfLimits(robj.clone()));
```

Comment:
Linting suggested doing this instead of what we had orignally.

Reply:
I agree with the lint